lib/capistrano/scm/tasks/git.rake

```diff
@@ -1,16 +1,7 @@
-namespace :git do
-
-  def strategy
-    @strategy ||= Capistrano::Git.new(self, fetch(:git_strategy, Capistrano::Git::DefaultStrategy))
-  end
-
-  set :git_environmental_variables, ->() {
-    {
-      git_askpass: "/bin/echo",
-      git_ssh:     "#{fetch(:tmp_dir)}/#{fetch(:application)}/git-ssh.sh"
-    }
-  }
+# This trick lets us access the Git plugin within `on` blocks.
+git = self
```

Comment:
This might be better named as git_plugin... git can mean a lot of things in here.